### PR TITLE
New CoreAudioReader using the AVAudioFile API as it's quicker

### PR DESCRIPTION
**Pull request by cthom055**

### CODE REVIEW

File: modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp [hunk 1]
```diff
@@ -383,229 +387,155 @@ struct CoreAudioFormatMetatdata
...
-            }
-        }
+        reusableBuffer = [[AVAudioPCMBuffer alloc] initWithPCMFormat:audioFile.processingFormat frameCapacity:InitialReusuableBufferSize];
+        
+        ok = true;
     }
 
-    ~CoreAudioReader() override
+    ~CoreAudioReader()
     {
```
Comment (davens) [inline modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp]: no cleanup req at all?

File: modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp [hunk 2]
```diff
@@ -383,229 +387,155 @@ struct CoreAudioFormatMetatdata
...
 
-            if ((int) numFramesToRead < numThisTime)
-            {
-                numThisTime = (int) numFramesToRead;
-                numBytes    = (size_t) numThisTime * sizeof (float);
-            }
+        // Also mostly copied from previous JUCE Reader
+        for (int i = numDestChannels; --i >= 0;)
+        {
+            auto* dest = destSamples[(i < static_cast<int>(numChannels) ? channelMap[i] : i)];
```
Comment (davens) [inline modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp:479]: const?

File: modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp [hunk 3]
```diff
@@ -383,229 +387,155 @@ struct CoreAudioFormatMetatdata
...
+        usesFloatingPointData = true;
 
-                        if (fileLayout.size() == static_cast<int> (numChannels))
-                        {
-                            hasLayout = true;
-                            channelSet = fileLayout;
-                        }
-                    }
-                }
+        auto channelLayout = audioFile.fileFormat.channelLayout.layout;
```
Comment (davens) [inline modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp]: const?

File: modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp [hunk 4]
```diff
@@ -383,229 +387,155 @@ struct CoreAudioFormatMetatdata
...
+            auto fileLayout = juce::CoreAudioLayouts::fromCoreAudio(*channelLayout);
+            AudioChannelSet channelSet;
 
-        return AudioFormatReader::getChannelLayout();
+            if (fileLayout.size() == static_cast<int>(numChannels))
+            {
+                channelSet = fileLayout;
+            }
+            
+            auto caOrder = juce::CoreAudioLayouts::getCoreAudioLayoutChannels(*channelLayout);
```
Comment (davens) [inline modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp]: const?

**Review by davens — APPROVED** [review]

File: modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp (hunk 2 above)
Comment (cthom055) [reply modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp:479]: dest can't be as it's getting stuff copied into it

**Pull request merged**